com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs

```diff
@@ -21,19 +21,42 @@ namespace UnityEditor.VFX
     [InitializeOnLoad]
     class VFXGraphPreprocessor : AssetPostprocessor
     {
-        void OnPreprocessAsset()
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
         {
-            bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
-            if (isVFX)
+            List<string> assetToReimport = null;
+
+            foreach (var assetPath in importedAssets)
             {
-                VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
-                if (resource == null)
-                    return;
-                VFXGraph graph = resource.graph as VFXGraph;
-                if (graph != null)
-                    graph.SanitizeForImport();
-                else
-                    Debug.LogError("VisualEffectGraphResource without graph");
+                bool isVFX = VisualEffectAssetModicationProcessor.HasVFXExtension(assetPath);
+                if (isVFX)
+                {
+                    VisualEffectResource resource = VisualEffectResource.GetResourceAtPath(assetPath);
+                    if (resource == null)
+                        return;
+                    VFXGraph graph = resource.graph as VFXGraph;
+                    if (graph != null)
+                    {
+                        bool wasGraphSanitized = graph.sanitized;
+                        graph.SanitizeForImport();
```

Comment:
I think this can throw? we may want to catch exception properly here?

Reply:
Good point : since this function is shared with other asset import, we should catch, log & safely continue the other import.

Reply:
Done at https://github.com/Unity-Technologies/Graphics/pull/4971/commits/f5a6be58f9efba2b28dfb6203d9e904fb847e262